sdc/functions/numpy_like.py

```diff
@@ -472,3 +475,187 @@ def nanprod_impl(a):
+            c *= v
+            out[idx] = c
+        return out
+
+    return array_cumprod_impl
+
+
+def cumsum(a):
+    pass
+
+
+def nancumsum(a):
+    pass
+
+
+def get_pool_size():
+    return numba.config.NUMBA_NUM_THREADS
+
+
+@sdc_overload(get_pool_size)
+def get_pool_size_overload():
+    pool_size = get_pool_size()
+
+    def get_pool_size_impl():
+        return pool_size
+
+    return get_pool_size_impl
+
+
+def get_chunks(size, pool_size=0):
+    if pool_size == 0:
+        pool_size = get_pool_size()
+
+    chunk_size = size//pool_size + 1
+
+    Chunk = namedtuple('start', 'stop')
+
+    chunks = []
+
+    for i in range(pool_size):
+        start = min(i*chunk_size, size)
+        stop = min((i + 1)*chunk_size, size)
+        chunks.append(Chunk(start, stop))
+
+    return chunks
+
+
+@sdc_overload(get_chunks)
+def get_chunks_overload(size, pool_size=0):
+    Chunk = namedtuple('Chunk', ['start', 'stop'])
+
+    def get_chunks_impl(size, pool_size=0):
+        if pool_size == 0:
+            pool_size = get_pool_size()
+
+        chunk_size = size//pool_size + 1
+
+        chunks = []
+
+        for i in range(pool_size):
+            start = min(i*chunk_size, size)
+            stop = min((i + 1)*chunk_size, size)
+            chunk = Chunk(start, stop)
+            chunks.append(chunk)
+
+        return chunks
+
+    return get_chunks_impl
+
+
+@sdc_overload(cumsum)
+def np_cumsum(arr):
+    if not isinstance(arr, types.Array):
+        return
+
+    retty = arr.dtype
+    zero = retty(0)
+
+    def cumsum_impl(arr):
+        chunks = get_chunks(len(arr))
+        partial_sum = numpy.zeros(len(chunks), dtype=retty)
+        result = numpy.empty_like(arr)
+
+        for i in prange(len(chunks)):
+            chunk = chunks[i]
+            partial = zero
+            for j in range(chunk.start, chunk.stop):
+                partial += arr[j]
+                result[j] = partial
+            partial_sum[i] = partial
+
+        for i in prange(len(chunks)):
+            prefix = sum(partial_sum[0:i])
+            chunk = chunks[i]
+            for j in range(chunk.start, chunk.stop):
+                result[j] += prefix
+
+        return result
+
+    return cumsum_impl
+
+
+@sdc_overload(nancumsum)
+def np_nancumsum(arr):
+    if not isinstance(arr, types.Array):
+        return
+
+    if isinstance(arr.dtype, (types.Boolean, types.Integer)):
+        # dtype cannot possibly contain NaN
+        return lambda arr: cumsum(arr)
+    else:
+        retty = arr.dtype
+        is_nan = get_isnan(retty)
+        zero = retty(0)
+
+        def nancumsum_impl(arr):
+            chunks = get_chunks(len(arr))
+            partial_sum = numpy.zeros(len(chunks), dtype=retty)
+            result = numpy.empty_like(arr)
+
+            for i in prange(len(chunks)):
+                chunk = chunks[i]
+                partial = zero
+                for j in range(chunk.start, chunk.stop):
+                    if ~is_nan(arr[j]):
```

Comment:
in case if `arr[j]` is `nan`, we should put `nan` to `result[j]`.
So, we can rewrite it in this way:
```python
result[j] = partial + arr[j]
if ~isnan(result[j]):
    partial = result[j]
```


And for consistency we could rewrite `cumsum` as:
```python
result[j] = partial + arr[j]
partial = result[j]
```

Reply:
With skipna=True, Pandas inserts NaN in place of NaN but NumPy treats NaN as 0 and inserts partial values in place of NaN.